mergify_engine/web/api/queues.py

```diff
@@ -317,5 +358,297 @@ async def repository_queues(
+            await qf.save()
+            return QueueFreezeResponse(
+                queue_freeze=[
+                    QueueFreeze(
+                        name=qf.name,
+                        reason=qf.reason,
+                        user_app_name=qf.user_app_name,
+                        freeze_date=qf.freeze_date,
+                    )
+                ],
+                description=f'The queue "{queue_name}" is already frozen. Its data have been successfully updated.',
+            )
+
+        return QueueFreezeResponse(
+            queue_freeze=[
+                QueueFreeze(
+                    name=qf.name,
+                    reason=qf.reason,
+                    user_app_name=qf.user_app_name,
+                    freeze_date=qf.freeze_date,
+                )
+            ],
+            description=f'The queue "{queue_name}" is already frozen.',
+        )
+
+
+@router.delete(
+    "/repos/{owner}/{repository}/queue/{queue_name}/unfreeze",  # noqa: FS003
+    summary="Unfreezes merge queue",
+    description="Unfreezes the specified merge queue",
+    status_code=204,
+)
+async def delete_queue_freeze(
+    owner: github_types.GitHubLogin = fastapi.Path(  # noqa: B008
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    queue_name: rules.QueueName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the queue"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+) -> None:
+
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token as access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+
+        qf = await queue_freeze.QueueFreeze.get(repository_ctxt, queue_name)
+        if qf is None:
+            raise fastapi.HTTPException(
+                status_code=404,
+                detail=f'The queue "{queue_name}" does not exists or is not currently frozen.',
+            )
+
+        await qf.delete()
```

Comment:
```suggestion
        qf = await queue_freeze.QueueFreeze(repository_ctxt, name=queue_name)
        if not await qf.delete()
            raise fastapi.HTTPException(
                status_code=404,
                detail=f'The queue "{queue_name}" does not exists or is not currently frozen.',
            )
```

This avoids doing a `HGET` + `HDEL` and just executes a `HDEL` for the same result, avoiding race conditions and saving time!